SERVER-1105 | Merge server and cloud runner docs (#5420)
circleci/circleci-docs: jekyll/_cci2/runner-installation.adoc

:toc: macro
:toc-title:

NOTE: CircleCI runner is available on the https://circleci.com/pricing[Scale

> Plan] and with https://circleci.com/pricing/server/[Server]. Please reach out

Reviewer: ```suggestion
plan] and with https://circleci.com/pricing/server/[server]. Please reach out
```

| `platform=darwin/arm64`
|===

Next, set the `circleci-launch-agent` version. On cloud, we select the latest version by

> querying for our releases for it and on server, we must select the version

Reviewer: This sentence is confusing me... "On cloud, we query our releases for the latest version..." maybe?

Reviewer 2: Is it more direct to say something like "Runners on cloud auto-update to the latest supported versions.  For Server, specific runner versions are validated for interoperability and runners do not auto-update"

That gets around the use of "we", if nothing else :)

Reviewer: I'll add this change

> querying for our releases for it and on server, we must select the version

Reviewer: ```suggestion
querying for our releases for it. On server, we must select the version
```

> Next, set the `circleci-launch-agent` version. On cloud, we select the latest version by
> querying for our releases for it and on server, we must select the version
> appropriate for our version of server. A table of server `circleci-launch-agent` versions
> can be found https://circleci.com/docs/2.0/runner-server/#version[here].

Reviewer: ```suggestion
Next, set the `circleci-launch-agent` version. Runners on cloud auto-update to the latest supported versions. For server, specific runner versions are validated for interoperability and runners do not auto-update. A table of server `circleci-launch-agent` versions
can be found https://circleci.com/docs/2.0/runner-server/#version[here].
```

For cloud, you can run the following:
```bash
export base_url="https://circleci-binary-releases.s3.amazonaws.com/circleci-launch-agent"
export agent_version=$(curl "${base_url}/release.txt")
```

> For server, run the following:

Reviewer: ```suggestion
For server v3.1.0, run the following:
```
